pkg/ovs/ovsconfig/ovs_client.go

```diff
@@ -879,3 +879,50 @@ func (br *OVSBridge) getHardwareOffload() (bool, Error) {
 func (br *OVSBridge) GetOVSDatapathType() OVSDatapathType {
 	return br.datapathType
 }
+
+// SetInterfaceTypeInternal modifies the OVS Interface type as "internal", and return until the ofport is allocated by
+// OVS. This function is used on Windows when the Pod interface is created after the OVS port creation.
+func (br *OVSBridge) SetInterfaceTypeInternal(name string) (int32, Error) {
+	// Update Interface type, and the caller ensures the host Interface exists.
+	tx1 := br.ovsdb.Transaction(openvSwitchSchema)
+	tx1.Update(dbtransaction.Update{
+		Table: "Interface",
+		Where: [][]interface{}{{"name", "==", name}},
+		Row: map[string]interface{}{
+			"type": "internal",
+		},
+	})
+	_, err, temporary := tx1.Commit()
+	if err != nil {
+		klog.Error("Transaction failed: ", err)
+		return 0, NewTransactionError(err, temporary)
+	}
+
+	// Wait until OVS allocated the ofport to the Interface.
```

Comment:
It seems the following code is duplicate with `GetOFPort`. Why don't we just call `GetOFPort` on caller side?

Reply:
That's still one line difference, could we share the code? For example, make `GetOFPort` take one argument `waitUntilValid`, set it `true` for the new scenario on client side.